tests/unit/src/test/scala/tests/codeactions/FilterMapToCollectCodeActionSuite.scala

```diff
@@ -0,0 +1,277 @@
+       |  case x if x > 2 =>
+       |    x * 2
+       |}
+       |}
+       |""".stripMargin
+
+  check(
+    "cursor-on-filter",
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  numbers.fil<<>>ter(x => x > 2).map(x => x * 2)
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toBraces("filter")}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |${FilterMapToCollectCodeAction.title}
+        |""".stripMargin,
+    simpleOutput,
+    selectedActionIndex = 2,
+  )
+
+  check(
+    "cursor-on-filter-newline",
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  numbers.fil<<>>ter{
+       |    x => x > 2
+       |  }.map{
+       |    x => x * 2
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toParens("filter")}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |${FilterMapToCollectCodeAction.title}
+        |""".stripMargin,
+    simpleOutput,
+    selectedActionIndex = 2,
+  )
+
+  check(
+    "cursor-on-map",
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  numbers.filter(x => x > 2).m<<>>ap(x => x * 2)
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toBraces("map")}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |${FilterMapToCollectCodeAction.title}
+        |""".stripMargin,
+    simpleOutput,
+    selectedActionIndex = 2,
+  )
+
+  check(
+    "higher-order-function",
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  def check = (x: Int) => x > 2
+       |  def double = (x: Int) => x * 2
+       |  numbers.fil<<>>ter(check).map(double)
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toBraces("filter")}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |${FilterMapToCollectCodeAction.title}
+        |""".stripMargin,
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  def check = (x: Int) => x > 2
+       |  def double = (x: Int) => x * 2
+       |  numbers.collect {
+       |  case x if check(x) =>
+       |    double(x)
+       |}
+       |}
+       |""".stripMargin,
+    selectedActionIndex = 2,
+  )
+
+  check(
+    "higher-order-function-rename-filter-argument",
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  def double = (x: Int) => x * 2
+       |  numbers.fil<<>>ter(x => x > 0).map(double)
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toBraces("filter")}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |${FilterMapToCollectCodeAction.title}
+        |""".stripMargin,
+    """|object Main {
+       |  val numbers = List(1, 2, 3, 4)
+       |  def double = (x: Int) => x * 2
+       |  numbers.collect {
+       |  case x if x > 0 =>
```

Comment:
We should probably add indentation here before case

Reply:
what would be the base way to inolement it? i wasn't able to find formatting functions and templating feels fragile.

Reply:
applied in f4d2a4a5b37ef34fd4acd75dd5ddb81a608e4668 (force pushed)